template issue

Hi, insalled pmz, configured `~/.pmz.yaml` to mention `ztldir: /Users/<username>/zettle` and ran `pmz new`.  It gives the following error.

```
Failed to write template to new note, but it should exist for you.
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x1102e19]

goroutine 1 [running]:
text/template.errRecover(0xc000117ba0)
	/usr/local/go/src/text/template/exec.go:163 +0x15b
panic({0x12aee20, 0x152e290})
	/usr/local/go/src/runtime/panic.go:1038 +0x215
text/template.(*Template).execute(0x0, {0x135a440, 0xc0000ae0c8}, {0x12b2740, 0x156dd80})
	/usr/local/go/src/text/template/exec.go:214 +0x239
text/template.(*Template).Execute(...)
	/usr/local/go/src/text/template/exec.go:200
github.com/gsilvapt/pmz/cmd.writeToNewNote(0xc0000b4198, {0x0, 0x0})
	/Users/<username>/go/pkg/mod/github.com/gsilvapt/pmz@v0.0.0-20211024185000-aa021a65b9a0/cmd/new.go:90 +0xf1
github.com/gsilvapt/pmz/cmd.glob..func1(0x1534c00, {0x12ec824, 0x0, 0x0})
	/Users/<username>/go/pkg/mod/github.com/gsilvapt/pmz@v0.0.0-20211024185000-aa021a65b9a0/cmd/new.go:52 +0x354
github.com/spf13/cobra.(*Command).execute(0x1534c00, {0x156c970, 0x0, 0x0})
	/Users/<username>/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:860 +0x5f8
github.com/spf13/cobra.(*Command).ExecuteC(0x1534e80)
	/Users/<username>/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:974 +0x3bc
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/<username>/go/pkg/mod/github.com/spf13/cobra@v1.2.1/command.go:902
github.com/gsilvapt/pmz/cmd.Execute()
	/Users/<username>/go/pkg/mod/github.com/gsilvapt/pmz@v0.0.0-20211024185000-aa021a65b9a0/cmd/root.go:50 +0x25
main.main()
	/Users/<username>/go/pkg/mod/github.com/gsilvapt/pmz@v0.0.0-20211024185000-aa021a65b9a0/main.go:21 +0x17
```

It created `/Users/<username>/zettle/2021113316/README.md`, but well, it did not open the directory or the file.  I did not find anything in readme, though inspecting the source suggests it is expecting a template file for the zettle readme?  Can you please update the readme to suggest how it should be used?

--- a/.gitignore
+++ b/.gitignore
@@ -9,6 +9,7 @@ bin/
 
 # Test binary, built with `go test -c`
 *.test
+.pmz.yaml
 
 # Output of the go coverage tool, specifically when used with LiteIDE
 *.out

--- a/.pmz.yaml.example
+++ b/.pmz.yaml.example
@@ -1,5 +1,6 @@
 editor: your editor path. Example: /usr/bin/vim
 ztldir: your zettel directory - it must exist! Example: /home/<user>/zettel
+notetemplate: if you have a note template you would like to use. Example: /home/<user>/zettel_tmpl.txt
 gitrepo: a link to a remote git repository. Example: git@gitlab.com:<user>/zettel.git
 gituser: your username in that remote git repositories server. 
 repotoken: the access token you generated in that server to grant you programmatic access. Ensure it provides at least read and write access to repositories.

--- a/README.md
+++ b/README.md
@@ -22,6 +22,16 @@ you will need to update to your own use case. Make a copy in your home directory
 * `pmz search` allows searching for keywords in notes' titles. Afterwards, user can either open or request more to 
 display file contents.
 
+## Templates  
+The program supports a custom template for a new note. It can have the format you want, as long as it contains the 
+`{{.Title}}` variable. If in doubt of what that looks like, please check [the template that exists](./templates/new_note) 
+in this repository. 
+
+You are free to use it too, either by cloning this project or by copying it manually to file in your system. 
+
+To use your custom template, please provide its full path in the configuration file. Check the 
+[sample configuration file](./.pmz.yaml.example) for an example.
+
 
 # CONTRIBUTING  
 When I started building this, I wanted a way to keep a [Zettelkasten](https://zettelkasten.de/) simple and bloat free - 

--- a/cmd/new.go
+++ b/cmd/new.go
@@ -36,10 +36,13 @@ For example:
 
 <pmz new> will create a new note (README.md) in a directory with the current timestamp and opens it up in your $EDITOR.
 You will see the new directory and file created in the configured ZTLDIR.
+
+If a template is specified and you use the --title flag, it will try to insert the title in that template.
 `,
 	Run: func(cmd *cobra.Command, args []string) {
 		ztldir := viper.GetString("ztldir")
 		editor := viper.GetString("editor")
+		tmpl_path := viper.GetString("notetemplate")
 		title, _ := cmd.Flags().GetString("title")
 		toOpen, _ := cmd.Flags().GetBool("open")
 
@@ -49,7 +52,10 @@ You will see the new directory and file created in the configured ZTLDIR.
 		f := createZettelEntry(ztldir, formattedTs)
 		defer f.Close()
 
-		writeToNewNote(f, title)
+		if tmpl_path != "" {
+			writeTmplToNote(f, title, tmpl_path)
+		}
+
 		if toOpen {
 			OpenFile(f.Name(), editor)
 		}
@@ -74,22 +80,23 @@ func createZettelEntry(ztldir, dirname string) *os.File {
 	return f
 }
 
-// writeToNewNote writes the template into the new file. Handles errors gracefully.
-func writeToNewNote(f *os.File, title string) {
+// writeTmplToNote writes the template into the new file. Handles errors gracefully.
+func writeTmplToNote(f *os.File, title, tmplPath string) {
 	tmplData := struct {
 		Title string
 	}{
 		Title: title,
 	}
 
-	tmpl, err := template.ParseFiles("templates/new_note")
+	tmpl, err := template.ParseFiles(tmplPath)
 	if err != nil {
-		Logger.Error("failed to write template to new note, but it should exist for you.")
+		Logger.Error("template file might not exist; not writing to new note.")
 	}
 
 	if err := tmpl.Execute(f, tmplData); err != nil {
-		Logger.Info("Failed to write template to new note, but it should exist for you.")
+		Logger.Info("Failed to write template to new note, but the note should exist.")
 	}
+	return
 }
 
 func init() {

--- a/cmd/root.go
+++ b/cmd/root.go
@@ -21,18 +21,18 @@ import (
 
 	"github.com/gsilvapt/pmz/internal/logs"
 	"github.com/spf13/cobra"
-
 	"github.com/spf13/viper"
 )
 
 var (
-	cfgFile   string
-	ztldir    string
-	editor    string
-	gitrepo   string
-	gituser   string
-	repotoken string
-	Logger    *logs.Log
+	cfgFile      string
+	ztldir       string
+	notetemplate string
+	editor       string
+	gitrepo      string
+	gituser      string
+	repotoken    string
+	Logger       *logs.Log
 )
 
 // rootCmd represents the base command when called without any subcommands
@@ -40,7 +40,11 @@ var rootCmd = &cobra.Command{
 	Use:   "pmz",
 	Short: "Poor Man Zettelkasten CLI",
 	Long: `This is a simple CLI application to help users maintain a Zettelkasten.
-It provides methods to add, search and save your changes into a git repository.`,
+It provides methods to add, search and save your changes into a git repository.
+
+Full documentation can be found in the project's README: https://github.com/gsilvapt/pmz
+`,
+	Version: "0.1",
 	// Uncomment the following line if your bare application
 	// has an action associated with it:
 	// Run: func(cmd *cobra.Command, args []string) { },
@@ -61,18 +65,18 @@ func init() {
 	// will be global for your application.
 	rootCmd.PersistentFlags().StringVar(&cfgFile, "config", "", "config file (default is $HOME/.pmz.yaml)")
 	rootCmd.PersistentFlags().StringVar(&ztldir, "ztldir", "", "Zettelkasten main directory loaded from config.")
+	rootCmd.PersistentFlags().StringVar(&notetemplate, "notetemplate", "", "Path to template of a new note. Ensure it contains the variables specified in the documentation.")
 	rootCmd.PersistentFlags().StringVar(&editor, "editor", "", "Editor PATH loaded from config file.")
 	rootCmd.PersistentFlags().StringVar(&gitrepo, "gitrepo", "", "Zettelkasten git repository loaded from config.")
 	rootCmd.PersistentFlags().StringVar(&gituser, "gituser", "", "Git username loaded from config.")
 	rootCmd.PersistentFlags().StringVar(&repotoken, "repotoken", "", "Zettelkasten git repository token with **read** and **write** access")
 
-	// Cobra also supports local flags, which will only run
-	// when this action is called directly.
-	rootCmd.Flags().BoolP("toggle", "t", false, "Help message for toggle")
+	rootCmd.InitDefaultVersionFlag()
 
 	// Viper binding for global reach
 	viper.BindPFlag("ztldir", rootCmd.PersistentFlags().Lookup("ztldir"))
 	viper.BindPFlag("editor", rootCmd.PersistentFlags().Lookup("editor"))
+	viper.BindPFlag("notetemplate", rootCmd.PersistentFlags().Lookup("notetemplate"))
 	viper.BindPFlag("gitrepo", rootCmd.PersistentFlags().Lookup("gitrepo"))
 	viper.BindPFlag("gituser", rootCmd.PersistentFlags().Lookup("gituser"))
 	viper.BindPFlag("repotoken", rootCmd.PersistentFlags().Lookup("repotoken"))

--- a/templates/new_note
+++ b/templates/new_note
@@ -2,6 +2,5 @@
 
 # Summary
 
-
 # Notes
 

